### PR TITLE
Replace driver

### PR DESCRIPTION
**Pull request by yseto**
- use go.mongodb.org/mongo-driver
- fix test script

### CODE REVIEW

File: lib/mongodb.go [hunk 1]
```diff
@@ -322,22 +181,37 @@ func (m MongoDBPlugin) LabelPrefix() string {
...
 // Do the plugin
 func Do() {
-	optHost := flag.String("host", "localhost", "Hostname")
-	optPort := flag.String("port", "27017", "Port")
+	optHost := flag.String("host", "", "Hostname")
+	optPort := flag.String("port", "", "Port")
 	optUser := flag.String("username", "", "Username")
 	optPass := flag.String("password", os.Getenv("MONGODB_PASSWORD"), "Password")
 	optSource := flag.String("source", "", "authenticationDatabase")
+	optURL := flag.String("url", os.Getenv("MONGODB_URL"), "URL (exclusive)")
```
Comment (Arthur1) [inline lib/mongodb.go]: [may] This usage does not tell us what is exclusive with MONGO_URL.

File: lib/mongodb.go [hunk 2]
```diff
@@ -322,22 +181,37 @@ func (m MongoDBPlugin) LabelPrefix() string {
...
 	flag.Parse()
 
 	var mongodb MongoDBPlugin
 	mongodb.Verbose = *optVerbose
-	mongodb.URL = net.JoinHostPort(*optHost, *optPort)
+
+	mongodb.RawURL = *optURL
+
+	if *optURL != "" && (*optHost != "" || *optPort != "" || *optUser != "" || *optPass != "" || *optSource != "") {
+		fmt.Println("-url is exclusive.")
```
Comment (Arthur1) [inline lib/mongodb.go]: [may] Same as the point about the usage of URL. How about the following expression?

> -url option is mutually exclusive with these options: -host, -port, -username, -password, and -source

Comment (Arthur1) [reply lib/mongodb.go]: This configuration can be set using the environment variable. However, this message only considers command line options.

File: lib/mongodb.go [hunk 3]
```diff
@@ -322,22 +181,37 @@ func (m MongoDBPlugin) LabelPrefix() string {
...
+	if *optHost == "" {
+		*optHost = "localhost"
+	}
+	if *optPort == "" {
+		*optPort = "27017"
+	}
+	mongodb.URL = fmt.Sprintf("mongodb://%s", net.JoinHostPort(*optHost, *optPort))
 	mongodb.Username = *optUser
 	mongodb.Password = *optPass
 	mongodb.Source = *optSource
```
Comment (Arthur1) [inline lib/mongodb.go:213]: [q] Is this the expected behavior that these parameters are set here even if the rawURL is non-empty?

**Review by Arthur1 — COMMENTED** [review]
It seems like a good idea to switch to another driver. I pointed out the command line interface.

File: lib/mongodb.go (hunk 3 above)
Comment (yseto) [reply lib/mongodb.go:213]: This was implemented to match the behavior of the previous version, as intended.

File: lib/mongodb.go (hunk 2 above)
Comment (yseto) [reply lib/mongodb.go]: applied your suggestion message.

File: README.md [hunk 4]
```diff
@@ -6,7 +6,7 @@ MongoDB custom metrics plugin for mackerel.io agent.
 ## Synopsis
 
 ```shell
-mackerel-plugin-mongodb [-host=<host>] [-port=<port>] [-username=<username>] [-password=<password>] [-tempfile=<tempfile>] [-source=<authenticationDatabase>]
+mackerel-plugin-mongodb [-host=<host>] [-port=<port>] [-username=<username>] [-password=<password>] [-tempfile=<tempfile>] [-source=<authenticationDatabase>] [-url=<mongodb://.....>]
```
Comment (Arthur1) [inline README.md]: -url option is mutually exclusive with other options. How about splitting the synopsis into two parts?

````
```shell
mackerel-plugin-mongodb [-host=<host>] [-port=<port>] [-username=<username>] [-password=<password>] [-tempfile=<tempfile>] [-source=<authenticationDatabase>]
```

```shell
mackerel-plugin-mongodb [-url=<mongodb://.....>] [-tempfile=<tempfile>]
```
````

**Review by Arthur1 — APPROVED** [review]
Looks great to us.

**Pull request merged**